File: SAMSUI/src/app/Model/building/building-segment.ts
import { BaseModel } from '../base/baseModel';

export class BuildingSegmentModel extends BaseModel{
     
    public orgId ?: number;

    public locationName ?: string;

    public locationId ?: number;

    public buildingSegmentId ?: number;

    public segmentName ?: string;

    public segmentDesc ?: string;

    public active ?: boolean;
    
    public createdBy ?: string;
    
    public createdDt ?: Date;
    
    public updatedBy ?: string;
    
    public updatedDt ?: Date;
    
	public pageNumber ?: number;

	public recordsPerPage ?: number;

	public updatedDtDisp ?: string;

	public createdDtDisp ?: string;
	
	public logInUserOrgId ?: number;
	
	public logInUserLocId ?: number;
	
    public logInUserId ?: number;
    
    public activeDisp ?: boolean;

    public activeDisplay ?: string

}

File: sams-backend/src/main/java/com/sams/entity/BuildingSegment.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_building_segment", schema = "base")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class BuildingSegment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "building_segment_id")
    private Long buildingSegmentId;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "location_name")
    private String locationName;

    @Column(name = "segment_name")
    private String segmentName;

    @Column(name = "segment_desc")
    private String segmentDesc;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

}